### PR TITLE
Asset role field

### PR DESCRIPTION
**Pull request by philvarner**
**Related Issue(s):** n/a

**Proposed Changes:**

* add a field rel to Asset to be used for the purpose of the asset, similarly to rel in Link

PR Checklist:

**PR Checklist:**

- [X] This PR has **no** breaking changes.
- [X] I have added my changes to the [CHANGELOG](https://github.com/radiantearth/stac-spec/blob/dev/CHANGELOG.md) **or** a CHANGELOG entry is not required.
- [X] API only: I have run `npm run generate-all` to update the [generated OpenAPI files](https://github.com/radiantearth/stac-spec/blob/dev/api-spec/README.md#openapi-definitions).

### CODE REVIEW

File: api-spec/STAC-extensions.yaml [hunk 1]
```diff
@@ -1779,6 +1779,10 @@ components:
             type: string
             description: Media type of the asset
             example: image/png
+          purpose:
```
Comment (m-mohr) [inline api-spec/STAC-extensions.yaml]: It's still purpose here.

File: CHANGELOG.md [hunk 2]
```diff
@@ -6,6 +6,13 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 
 ## Unreleased
 
+### Changed
+- Added attribute `purpose` to Asset, to be used similarly to Link `rel`
```
Comment (m-mohr) [inline CHANGELOG.md]: It's still purpose here.

File: item-spec/item-spec.md [hunk 3]
```diff
@@ -167,6 +167,19 @@ or streamed. It is allowed to add additional fields.
...
 | title      | string | The displayed title for clients and users.                                            |
 | type       | string | [Media type](#media-types) of the asset.                                              |
+| rel        | string | The purpose of the asset, for example `thumbnail`.                                    |
+
+**rel** is intended to be a tag that adds semantic usage information to an asset, similarly to the use of `rel` in Link, even though it is not 
+really a relationship as the `rel` name implies.
+ 
+Examples of this are:
+
+* **thumbnail** An asset that represents a thumbnail of the item, typically a true color image (for items with assets
```
Comment (m-mohr) [inline item-spec/item-spec.md]: Maybe better make this a table.

**Comment by m-mohr** [comment]
So: rel or role? If it's going to be role I'd remove all references to the link rel and IANA lists...

File: item-spec/item-spec.md (hunk 3 above)
Comment (philvarner) [reply item-spec/item-spec.md]: done

**Comment by m-mohr** [comment]
Related issues: #198 , #309, #308

As discussed on the sprint, we should think about other recommended types. Some discussions on this in #308

File: item-spec/item-spec.md [hunk 4]
```diff
@@ -167,6 +167,20 @@ or streamed. It is allowed to add additional fields.
...
+ 
+#### Asset Role Types
+
+Like the Link `rel` field, the `role` field can be given any value.  These are a few standardized role names.
+
+| Role Name | Description                                                                           |
+| --------- | ------------------------------------------------------------------------------------- |
+| thumbnail |  An asset that represents a thumbnail of the item, typically a true color image (for items with assets in the visible wavelengths), lower-resolution (typically smaller 600x600 pixels), and typically a JPEG or PNG (suitable for display in a web browser). Multiple assets may have this purpose, but it recommended that the `type` and `role` be unique tuple. For example, Sentinel-2 L2A provides thumbnail images in both JPEG and JPEG2000 formats, and would be distinguished by their media types. |
+| overview  | An asset that represents a possibly larger view than the thumbnail of the Item , for example, a true color composite of multi-band data. |
+| metadata  | A metadata sidecar file describing the data in this item, for example the Landsat-8 MTL file. |
```
Comment (hgs-truthe01) [inline item-spec/item-spec.md]: Should we add an "analytic" role or something similar? What do we tag the full image files with?

Comment (m-mohr) [reply item-spec/item-spec.md]: See #308 for related discussions (primary? analytic?)

Comment (hgs-truthe01) [reply item-spec/item-spec.md]: Thanks! I approved as is, so I'm happy to leave discussion there

File: item-spec/examples/digitalglobe-sample.json [hunk 5]
```diff
@@ -59,7 +59,8 @@
       "href": "https://15NOV09180446-M1BS-056823192010_01_P002.RPB"
     },
     "thumbnail": {
-      "href": "https://15NOV09180446-M1BS-056823192010_01_P002.JPG"
+      "href": "https://15NOV09180446-M1BS-056823192010_01_P002.JPG".
```
Comment (matthewhanson) [inline item-spec/examples/digitalglobe-sample.json]: This needs to be a comma

**Comment by matthewhanson** [comment]
This looks good to me and I think the three roles are fine to start with for now.
Just one error in the DG sample

File: item-spec/examples/digitalglobe-sample.json (hunk 5 above)
Comment (philvarner) [reply item-spec/examples/digitalglobe-sample.json]: fixed.

**Comment by philvarner** [comment]
I created a new PR to have this allow multiple roles https://github.com/radiantearth/stac-spec/pull/655

**Pull request merged**